packages/polymarket-notifier/src/polymarketNotifier.js

```diff
@@ -0,0 +1,223 @@
+// This notifier is used to monitor and compare Optimistic Oracle contract proposals against the Polymarket API endpoint.
+
+const { Datastore } = require("@google-cloud/datastore");
+const datastore = new Datastore();
+const abi = require("./abi/abi");
+const Web3 = require("web3");
+const ethers = require("ethers");
+const uma = require("@uma/sdk");
+const { getAddress } = require("@uma/contracts-node");
+
+const web3 = new Web3(process.env.CUSTOM_NODE_URL);
+const provider = new ethers.providers.WebSocketProvider(process.env.CUSTOM_NODE_URL);
+
+const polymarketContract = "0xCB1822859cEF82Cd2Eb4E6276C7916e692995130";
+const earlyExpiryResponse = "-57896044618658097711785492504343953926634992332820282019728792003956564819968";
+
+class PolymarketNotifier {
+  /**
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} networker Used to send the API requests.
+   * @param {Function} getTime Returns the current time.
+   * @param {String} apiEndpoint API endpoint to monitor.
+   * @param {Integer} maxTimeAfterProposal Period in seconds to look for past proposals.
+   */
+  constructor({ logger, networker, getTime, apiEndpoint, maxTimeAfterProposal }) {
+    this.logger = logger;
+    this.networker = networker;
+    this.getTime = getTime;
+    this.apiEndpoint = apiEndpoint;
+    this.maxTimeAfterProposal = maxTimeAfterProposal;
+  }
+
+  // Main function to check recent proposals against Polymarket API data.
+  async checkRecentProposals() {
+    this.logger.debug({
+      at: "PolymarketNotifier",
+      message: "Checking for past proposals",
+      apiEndpoint: this.apiEndpoint,
+      maxTimeAfterProposal: this.maxTimeAfterProposal,
+    });
+
+    const proposalEvents = [];
+    const proposalData = [];
```

Comment:
you should not push to constants like this. I know you can but ideally use `let`.